torch/_numpy/_ndarray.py

```diff
@@ -505,8 +505,13 @@ def array(obj, dtype=None, *, copy=True, order="K", subok=False, ndmin=0, like=N
     ):
         return obj
 
-    # lists of ndarrays: [1, [2, 3], ndarray(4)] convert to lists of lists
     if isinstance(obj, (list, tuple)):
+        # list of arrays *under torch.Dynamo*
+        if obj and all(isinstance(x, torch.Tensor) for x in obj):
+            return ndarray(torch.stack(obj))
```

Comment:
this should be a recursive call into `array` with all the args and kwargs it takes.

Reply:
Good catch, done.